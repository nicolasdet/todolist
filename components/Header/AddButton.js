import { View, Pressable, StyleSheet } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';

const AddButton = () => {
  const navigation = useNavigation();
  const onPress = () => {
    navigation.navigate('AddScreen');
  };

  return (
    <View style={styles.containerStyle}>
      <Pressable
        onPress={onPress}
        style={({ pressed }) => pressed && styles.pressed}
      >
        <Ionicons
          testID="Add-Icon"
          name="ios-add-outline"
          size={30}
          color="black"
        />
      </Pressable>
    </View>
  );
};

export default AddButton;

const styles = StyleSheet.create({
  containerStyle: {
    marginRight: 15,
    backgroundColor: 'white',
  },
  pressed: {
    opacity: 0.4,
  },
});
