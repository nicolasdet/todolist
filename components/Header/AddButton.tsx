import { View, Pressable, StyleSheet } from 'react-native';
import { RootStackNavigationProp } from '../navigation/NavigationTypes';
import { Ionicons } from '@expo/vector-icons';
import { useNavigation } from '@react-navigation/native';

const AddButton = () => {
  const navigation = useNavigation<RootStackNavigationProp>();
  const onPress = () => {
    navigation.navigate('AddScreen');
  };

  return (
    <View style={styles.containerStyle}>
      <Pressable
        onPress={onPress}
        style={({ pressed }) => pressed && styles.pressed}
      >
        <Ionicons
          testID="Add-Icon"
          name="ios-add-outline"
          size={30}
          color="black"
        />
      </Pressable>
    </View>
  );
};

export default AddButton;

const styles = StyleSheet.create({
  containerStyle: {
    marginRight: 15,
    backgroundColor: 'white',
  },
  pressed: {
    opacity: 0.4,
  },
});
