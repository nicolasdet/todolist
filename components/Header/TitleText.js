import { View, Text, StyleSheet } from 'react-native';

const TitleText = ({ text, extraStyles, extraContainerStyle }) => {
  return (
    <View style={[styles.container, extraContainerStyle]}>
      <Text style={[styles.textStyle, extraStyles]}>{text}</Text>
    </View>
  );
};

export default TitleText;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingLeft: 30,
  },
  textStyle: {
    fontSize: 20,
    fontWeight: 'bold',
    color: 'white',
  },
});
