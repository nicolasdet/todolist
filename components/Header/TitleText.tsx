import { View, Text, StyleSheet, ViewStyle } from 'react-native';

interface TitleTextProps {
	text: string;
	extraStyles?: ViewStyle ;
	extraContainerStyle?: ViewStyle;
}
const TitleText = ({ text, extraStyles, extraContainerStyle }: TitleTextProps) => {
  return (
    <View style={[styles.container, extraContainerStyle]}>
      <Text style={[styles.textStyle, extraStyles]}>{text}</Text>
    </View>
  );
};

export default TitleText;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    paddingLeft: 30,
  },
  textStyle: {
    fontSize: 20,
    fontWeight: 'bold',
    color: 'white',
  },
});
