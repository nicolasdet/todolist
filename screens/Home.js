import { View, StyleSheet } from 'react-native';
import Header from '../components/Header/Header';
import TodoList from '../components/TodoList/TodoList';
import { colors } from '../utils/colors';

const Home = () => {
  return (
    <View style={styles.container}>
      <View style={styles.headerContainer}>
        <Header />
      </View>
      <View style={styles.todoListContainer}>
        <TodoList />
      </View>
    </View>
  );
};

export default Home;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
  },
  headerContainer: {
    flex: 1,
    width: '100%',
    backgroundColor: colors.grey,
    justifyContent: 'center',
    alignItems: 'center',
  },
  todoListContainer: {
    flex: 2,
    width: '100%',
    backgroundColor: 'wheat',
  },
});
