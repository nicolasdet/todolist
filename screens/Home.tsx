import { View, StyleSheet, ImageBackground } from 'react-native';
import Header from '../components/Header/Header';
import TodoList from '../components/TodoList/TodoList';
import colors from '../utils/colors';
import { isTablet } from '../utils/deviceInfo';
import AddButton from '../components/AddButton/AddButton';

const Home = (): React.JSX.Element => {
  return (
    <ImageBackground
      source={require('../assets/beach.jpg')}
      resizeMode="cover"
      style={styles.image}
    >
      <View style={styles.container}>
        <View style={styles.headerContainer}>
          <Header />
        </View>
        <View style={styles.todoListContainer}>
          <TodoList />
        </View>
        <AddButton />
      </View>
    </ImageBackground>
  );
};

export default Home;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    paddingVertical: isTablet ? 0 : 30,
  },
  headerContainer: {
    flex: 1,
    width: '100%',
  },
  image: {
    flex: 1,
    width: '100%',
  },
  todoListContainer: {
    flex: 5,
    margin: 10,
  },
});
