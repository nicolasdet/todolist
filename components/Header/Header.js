import { View, Text, StyleSheet } from 'react-native';
import { colors } from '../../utils/colors';
import { isTablet } from '../../utils/deviceInfo';
import AddButton from './AddButton';
import TitleText from './TitleText';

const Header = () => {
  return (
    <View style={styles.container}>
      <TitleText text="Ma liste de taches" />
      <AddButton />
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    width: isTablet ? '50%' : '90%',
    backgroundColor: colors.blue,
    height: 50,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
  },
});
