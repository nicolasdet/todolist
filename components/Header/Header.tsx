import { View, StyleSheet } from 'react-native';
import AppText from '../../utils/Text.json';
import colors  from '../../utils/colors';
import { isTablet } from '../../utils/deviceInfo';
import TitleText from './TitleText';

const Header = () => {
  return (
    <View style={styles.container}>
		<View>
      		<TitleText text={AppText.Todo.Header.Title} />
	  		<TitleText text={AppText.Todo.Header.Hello} />
		</View>
    </View>
  );
};

export default Header;

const styles = StyleSheet.create({
  container: {
    width: "100%",
    height: isTablet ? 200 : 100,
	flexDirection: 'row',
	alignItems: 'center',
	justifyContent: 'space-between',
  },
});
