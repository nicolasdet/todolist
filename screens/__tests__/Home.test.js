import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native';
import Home from '../TodoHome';
import Header from '../../components/Header/Header';

// Mocks like this need to be configured at the top level
// of the test file, they can't be setup inside your tests.
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
  };
});

it('renders Home correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('show header Bar correctly', () => {
  const { getByText } = render(<Header />);
  const titleText = getByText('Ma liste de taches');
  expect(titleText).toBeTruthy();
});

test('Icon is visible from Home page', () => {
  const { queryByTestId } = render(<Home />);
  const icon = queryByTestId('Add-Icon');
  expect(icon).toBeTruthy();
});
