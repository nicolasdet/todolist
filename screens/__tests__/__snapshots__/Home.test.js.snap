// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Home correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#e1e5f2",
        "flex": 1,
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1f7a8c",
          "flexDirection": "row",
          "height": 50,
          "justifyContent": "center",
          "width": "90%",
        }
      }
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "paddingLeft": 30,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "bold",
              },
              undefined,
            ]
          }
        >
          Ma liste de taches
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "white",
            "marginRight": 15,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={false}
        >
          <Text />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "backgroundColor": "#e1e5f2",
        "flex": 3,
        "width": "100%",
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "marginTop": 20,
          "width": "100%",
        }
      }
    >
      <RCTScrollView
        data={[]}
        getItem={[Function]}
        getItemCount={[Function]}
        keyExtractor={[Function]}
        onContentSizeChange={[Function]}
        onLayout={[Function]}
        onMomentumScrollBegin={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={[]}
        viewabilityConfigCallbackPairs={[]}
      >
        <View />
      </RCTScrollView>
    </View>
  </View>
</View>
`;
